### DISCUSSION TITLE
About Text function, can we enforce a max length on format string argument?

### DISCUSSION THREAD

**Discussion by anderson-joyle** [microsoft/Power-Fx#897] [General]
The function `Text` takes a format arg[1] used to format the arg[0] (depending on arg[0] type).
For instance: `Text(DateTime(2022,12,1,23,45,59), "yY-m-d h:M:Ss")` result is `22-12-1 23:45:59`.

Is there any reason why we would allow more than a certain number of characters (let's say around 100)?

Limiting the number of characters for arg[1] would mitigate some possible issues.

**Comment by MikeStall** [comment]
I think limiting this is ok. If somebody has a use case where it ought to be longer, then we can revisit.

**Comment by cchannon** [comment]
+1

Hard to imagine any case where someone would need to produce a 100+ character formatted datetime string, but even if such a circumstance did exist, it is hardly a stretch to tell someone that the simply need to:

```
Concat(
    Text(
        DateTime(2022,12,1,23,45,59),
        "{Part 1 of my crazy string}"
    ),
    Text(
        DateTime(2022,12,1,23,45,59),
        "{Part 2 of my crazy string}"
    )
)
```

**Comment by anderson-joyle** [comment]
As per [PR 890](https://github.com/microsoft/Power-Fx/pull/890), arg1 (format string) is now limited to 100 characters.
